src/Propel/Runtime/Validator/Constraints/DateValidator.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * This file is part of the Propel package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license MIT License
+ */
+
+namespace Propel\Runtime\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraints\DateValidator as SymfonyDateValidator;
+use Symfony\Component\Validator\Constraint;
+
+class DateValidator extends SymfonyDateValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+
+        if ($value instanceof \DateTimeInterface) {
+            return;
+        }
```

Comment:
SF4.2 deprecated this entirely and removed in v5, but we use it everywhere internally.

In the 2.x series it should get majored out.

Reply:
Do we have this documented somewhere or as comment?

Reply:
There's the commit message for this change:

> 1f261c8c51b2bf9af8d15f9dff7012a31dda7bc1 Shim DateValidator to work with \Date objects as it did in sf4.

And the changelog in Sf:

https://github.com/symfony/validator/blob/master/CHANGELOG.md#500

Reply:
Let's hope the commit message isnt forgotten. it would be best to have some kind of deprecation flag or log of what should be done for 2.0 cleanup.

Reply:
We haven't discussed deprecating this behaviour from the API of propel itself.

We aren't deprecating it, we just use a tool that deprecated it.